draft-ietf-quic-transport.md

```diff
@@ -3192,6 +3234,48 @@ zero-length Destination Connection ID MUST treat receipt of a NEW_CONNECTION_ID
 frame as a connection error of type PROTOCOL_VIOLATION.
 
 
+## CONNECTION_ID_FINISHED Frame {#frame-connection-id-finished}
+
+An endpoint sends a CONNECTION_ID_FINISHED frame (type=0x1b) to indicate that it
+will no longer use a connection ID that was issued by its peer.  This also
+serves as a request to the peer to send additional connection IDs for future use
+(see {{connection-id}}).  New connection IDs can be delivered via the
+NEW_CONNECTION_ID frame ({{frame-new-connection-id}}).
+
+The CONNECTION_ID_FINISHED frame is as follows:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|   Length (8)  |          Connection ID (32..144)            ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+The fields are:
+
+Length:
+
+: An 8-bit unsigned integer containing the length of the connection ID.  Values
+  less than 4 and greater than 18 are invalid and MUST be treated as a
+  connection error of type PROTOCOL_VIOLATION.
```

Comment:
I don't think this is helpful. It's a PROTOCOL_VIOLATION to send an unknown CID here and so if you just enforce that, you will get the right behavior without a length check, as long as you check for 0.

Reply:
Retransmissions of CONNECTION_ID_FINISHED could arrive after the provider of connection IDs disposes of the old value.  That means that you can't generate errors unless you remember all previous connection IDs (this is the mirror-image of @kazuho's problem).  We can allow for PROTOCOL_VIOLATION if you like, but it might be hard to mandate.

Reply:
Well, the text below on line 3274 requires this.

Reply:
Yes, and that's a problem.

Reply:
Yeah, that’s a really good point, my initial feeling would be to not make it an error.

Reply:
Technically, you don't have to error on invalid CIDs.  You check whether the supplied CID is one you have associated with this connection; if so, remove it.  If the supplied CID is out-of-bounds, it won't be there.  If we feel we need an error case here, I think a MAY PV is sufficient.

Reply:
How about: 
```
An 8-bit unsigned integer containing the length of the connection ID.  Receipt of a value
   not previously issued to the peer as a valid connection ID MAY be treated as a
   connection error of type PROTOCOL_VIOLATION.
```

Reply:
Fixed in b54d0ce